### PR TITLE
docs(kubernetes_logs source): Add reference docs

### PR DESCRIPTION
**Pull request by binarylogic**
Supersedes https://github.com/timberio/vector/pull/3280 since we no longer host the website in this repo.

### CODE REVIEW

File: .meta/sources/kubernetes_logs.toml.erb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+use an env var by default, to be evaluated to a value provided by Kubernetes \
+at Pod deploy time.\
+"""
+
+[sources.kubernetes_logs.options.auto_partial_merge]
+type = "bool"
+common = false
+default = true
+description = """\
+Automatically merge partial messages into a signle event. \
```
Comment (jamtur01) [inline .meta/sources/kubernetes_logs.toml.erb:61]: single

File: .meta/sources/kubernetes_logs.toml.erb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+description = """\
+Configuration for how the events are annotated with Pod metadata.\
+"""
+
+[sources.kubernetes_logs.options.annotation_fields.children.pod_name]
+type = "string"
+default = "kubernetes.pod_name"
+sort = 1
+description = """\
+The field name to annotate Pod name as.\
```
Comment (jamtur01) [inline .meta/sources/kubernetes_logs.toml.erb:78]: I can't parse this sentence?

File: .meta/sources/kubernetes_logs.toml.erb [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+"""
+
+[sources.kubernetes_logs.options.annotation_fields.children.pod_namespace]
+type = "string"
+default = "kubernetes.pod_namespace"
+sort = 2
+description = """\
+The field name to annotate Pod namespace as.\
```
Comment (jamtur01) [inline .meta/sources/kubernetes_logs.toml.erb:86]: As above.

**Comment by binarylogic** [comment]
@MOZGIII would you mind making the changes above? (copied from https://github.com/timberio/vector/pull/3280)

Also, I'm realizing we'll need an entry into `.meta/installation/platforms.toml` as well. That'll drive the installation pages.

Thanks.

File: .meta/sources/kubernetes_logs.toml.erb (hunk 2 above)
Comment (MOZGIII) [reply .meta/sources/kubernetes_logs.toml.erb:78]: Would `The name of the event field to save Pod name as.` be better?

File: .meta/sources/kubernetes_logs.toml.erb [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+type = "bool"
+common = false
+default = true
+description = """\
+Automatically merge partial messages into a signle event. \
+Partial here is in respect to messages that were split by the Kubernetes \
+Container Runtime log driver.\
+"""
+
+[sources.kubernetes_logs.options.annotation_fields]
```
Comment (binarylogic) [inline .meta/sources/kubernetes_logs.toml.erb:66]: @MOZGIII I like that you nested field configuration under a key, but it deviates from other sources. For example, the `file` source uses root-level options like `host_key` and `file_key`. I don't think we should change it right now but wanted you to be aware. I think https://github.com/timberio/vector/issues/3463 is much better solution to all of this.

**Comment by MOZGIII** [comment]
> @MOZGIII would you mind making the changes above? (copied from #3280)

Sure.

> Also, I'm realizing we'll need an entry into `.meta/installation/platforms.toml` as well. That'll drive the installation pages.

Looks like we'll need to add a logo at the `vector-website`. I'll create a relevant PR.

**Comment by MOZGIII** [comment]
I'm having trouble checking the results of https://github.com/timberio/vector/pull/3555/commits/5108acedde755e9a8525b766ad7203d6832a2083 - does it look right?

File: .meta/sources/kubernetes_logs.toml.erb (hunk 3 above)
Comment (jamtur01) [reply .meta/sources/kubernetes_logs.toml.erb:86]: `Event field for Pod namespace.`

File: .meta/sources/kubernetes_logs.toml.erb (hunk 2 above)
Comment (jamtur01) [reply .meta/sources/kubernetes_logs.toml.erb:78]: Hmmm:

`Event field for Pod name.`

Or

`Event field that contains Pod name.`

Maybe?

Comment (MOZGIII) [reply .meta/sources/kubernetes_logs.toml.erb:78]: I like the `Event field for Pod name.`. Thx!

**Comment by MOZGIII** [comment]
Addressed the wording and typos

**Comment by MOZGIII** [comment]
Rebased the code to apply DCO signoff

**Review by jamtur01 — APPROVED** [review]
LGTM

**Review by MOZGIII — APPROVED** [review]
@binarylogic I think we can merge it!

**Pull request merged**